lib/daru/index.rb

```diff
@@ -330,5 +341,11 @@ def values
     def inspect
       "Daru::MultiIndex:#{self.object_id} (levels: #{levels}\nlabels: #{labels})"
     end
+
+    def factor_out
```

Comment:
I think this method will modify the MultiIndex data structures for the purpose of displaying.

Moreover, better displaying of the MultiIndex is an issue in itself and not a part of revamping index infra.